site/src/pages/release-notes/0.99.5.mdx

```diff
@@ -0,0 +1,170 @@
+- You can now register your server to or retrieve the <typeplural://Endpoint> from
+  a [Eureka](https://spring.io/guides/gs/service-registration-and-discovery/) registry server,
+  which gives nice integration with Spring Cloud. #707 #2701
+  - Use <type://EurekaUpdatingListener> for <typeplural://Server>:
+    ```java
+    Server server = Server.builder()...build();
+    server.addListener(EurekaUpdatingListener.of("https://my-eureka-registry.com/"));
+    // Automatically registers when the Server starts.
+    server.start().join();
+    // Automatically deregisters when the Server stops.
+    server.stop().join();
+    ```
+  - Use <type://EurekaEndpointGroup> for clients:
+    ```java
+    // Periodically retrieves the endpoint list from the registry.
+    EndpointGroup group = EurekaEndpointGroup.of("https://my-eureka-registry/");
+    group.whenReady().join();
+
+    // Sends a request to the endpoints, spreading the load w/ weighted round-robin.
+    WebClient client = WebClient.of(SessionProtocol.HTTP, group);
+    System.err.println(client.get("/some-resource")
+                             .aggregate().join()
+                             .contentUtf8());
+    ```
+- You can now use the gRPC client stubs generated by [`grpc-kotlin`](https://github.com/grpc/grpc-kotlin)
+  #2662 #2669
+- Added a new Thrift serialization format `TTEXT_NAMED_ENUM`. It is a variant of `TTEXT` that serializes
+  an `enum` as a string instead of an integer. This can be useful when you need to serialize a `TBase`
+  object into a human-readable JSON. #2706
+  ```java
+  ObjectMapper mapper = new ObjectMapper()
+  mapper.registerModule(new ThriftJacksonModule(/* useNamedEnum */ true));
+  String json = mapper.writeValueAsString(myTBaseStruct);
+  ```
+- Added two Spring Boot annotations that inject the current local Armeria port(s) - <type://@LocalArmeriaPort> and <type://@LocalArmeriaPorts>,
+  which is useful when you bind at an ephemeral port or more than one port: #2561 #2684
+  ```java
+  @Configuration
+  class MyConfiguration {
+      @LocalArmeriaPort(SessionProtocol.HTTP)
+      int httpPort;
+      @LocalArmeriaPort(SessionProtocol.HTTPS)
+      int httpsPort;
+      @LocalArmeriaPorts
+      List<Integer> ports;
+  }
+  ```
+
+
+## 📈 Improvements
+
+- <type://ZooKeeperUpdatingListener> now updates the registry with both host name and IP address by default,
+  so that the client does not have to do a DNS lookup. #2674 #2679
+  - The IP address is retrieved from `defaultNonLoopbackIpV4Address()` of <type://SystemInfo>.
+
+## 🛠️ Bug fixes
+
+- Fixed a bug where more than one thread can consume from a streaming message. #2687 #2683
+- You will not get a `NullPointerException` anymore when starting a <type://JettyService> with DEBUG level
+  logging enabled. #2688 #2691
+- PING interval now works as expected even if idle timeout is disabled. #2702
+- A server-side connection is now closed immediately when PING fails. #2699
```

Comment:
A client-side connection will be closed immediately too.
https://github.com/line/armeria/pull/2699/files#diff-e18570c3e534d61452c5286c7ec35704

Reply:
Thanks! Fixed.